R/paginate.R

```diff
@@ -13,13 +13,18 @@
 #'     that is used to describe the next page.
 #'
 #' @inheritParams req_perform
-#' @param next_request A callback function that takes a two arguments (the
-#'   original request and the response) and returns:
+#' @param next_request A callback function that returns a [request] to the next
+#'   page or `NULL` if there is no next page. It takes a three arguments:
 #'
-#'   * a new [request] to request the next page or
-#'   * `NULL` if there is no next page.
-#' @param n_pages A function that extracts the total number of pages from
-#'   the [response].
+#'   1. `req`: the original request
+#'   2. `resp`: the response of the current request
+#'   3. `body`: the result of the argument `body` below.
+#' @param body A function with one argument `resp` that parses the body of the
```

Comment:
This gets quite confusing. Maybe this should be renamed to `parse_body` or something. Any ideas for a good name?

Reply:
Maybe the argument to `req_paginate()` should be `parse_resp`? And then the argument to `next_request` would be `parsed`?